ipaserver/plugins/ldap2.py

```diff
@@ -419,7 +419,8 @@ def add_entry_to_group(self, dn, group_dn, member_attr='member', allow_same=Fals
                 modlist = [(a, b, self.encode(c))
                            for a, b, c in modlist]
                 self.conn.modify_s(str(group_dn), modlist)
-        except errors.DatabaseError:
+        except errors.DuplicateEntry:
+            # TYPE_OR_VALUE_EXISTS
             raise errors.AlreadyGroupMember()
 
```

Comment:
indeed it was a weird translation DB_ERROR -> AlreadyGroupMember

The patch looks good to me. ACK